src/main/java/com/amazon/opendistroforelasticsearch/security/ssl/transport/OpenDistroSecuritySSLNettyTransport.java

```diff
@@ -178,21 +200,42 @@ public void connect(ChannelHandlerContext ctx, SocketAddress remoteAddress, Sock
     protected class SSLClientChannelInitializer extends Netty4Transport.ClientChannelInitializer {
         private final boolean hostnameVerificationEnabled;
         private final boolean hostnameVerificationResovleHostName;
+        private final DiscoveryNode node;
+        private SSLConnectionTestResult connectionTestResult;
 
         public SSLClientChannelInitializer(DiscoveryNode node) {
+            this.node = node;
             hostnameVerificationEnabled = settings.getAsBoolean(
                     SSLConfigConstants.OPENDISTRO_SECURITY_SSL_TRANSPORT_ENFORCE_HOSTNAME_VERIFICATION, true);
             hostnameVerificationResovleHostName = settings.getAsBoolean(
                     SSLConfigConstants.OPENDISTRO_SECURITY_SSL_TRANSPORT_ENFORCE_HOSTNAME_VERIFICATION_RESOLVE_HOST_NAME, true);
+
+            connectionTestResult = SSLConnectionTestResult.SSL_AVAILABLE;
+            if (openDistroSSLConfig.isDualModeEnabled()) {
+                SSLConnectionTestUtil sslConnectionTestUtil = new SSLConnectionTestUtil(node.getAddress().getAddress(), node.getAddress().getPort());
```

Comment:
What is the reason to probe connection in `SSLClientChannelInitializer` constructor and not in `initChannel()`?

Reply:
performing these steps in initChannel() adds delays during the connection initialization, hence kept these in the constructor